src/recording-manager/RecordingManager.ts

```diff
@@ -34,23 +39,28 @@ export default class RecordingManager {
 
   async getRecordings(): Promise<Array<Recording>> {
     const items = await readdirAsync(this.#rootDir, { withFileTypes: true });
-    return items
-      .filter((item) => item.isDirectory())
-      .map(
-        (folder) =>
-          new Recording(
-            formatStorageNameForDate(folder.name),
-            new Date(formatStorageNameForDate(folder.name)),
-            path.join(this.#rootDir, folder.name),
-            this.#speechToText
-          )
-      )
-      .sort((x, y) => y.datetime.getTime() - x.datetime.getTime());
+    const recordings = await Promise.all(
+      items
+        .filter((item) => item.isDirectory())
+        .map(
+          async (folder) =>
+            new Recording(
+              folder.name,
+              (
+                await statAsync(path.join(this.#rootDir, folder.name))
+              ).birthtime,
+              path.join(this.#rootDir, folder.name),
+              this.#speechToText
+            )
+        )
+    );
+    return recordings.sort(
+      (x, y) => y.datetime.getTime() - x.datetime.getTime()
+    );
   }
 
   async createRecording(readStream: Readable): Promise<void> {
-    const newFolderName = formatDateForStorage(new Date());
-    const recordingDir = path.join(this.#rootDir, newFolderName);
+    const recordingDir = path.join(this.#rootDir, uuidv4());
```

Comment:
Would it be better to have a default name like ```New Recording 2``` when a new recording is created instead of having the UUID as the title?

Reply:
I agree that would be ideal, though there are some undesirable trade-offs. To achieve this, I would need to get a count of the current number of recording folders and increment that by 1, then create a new recording folder titled `Recording ${n+1}`. However, in the event where someone renames a previous recording to `Recording ${n+1}`, or worse so, a sequence of such recordings, I would be forced to introduce a loop on top of just counting the number of folders to find the next appropriate title, which can get very inefficient as the number of recordings grow. The fact that it's a trivial matter doesn't justify the cost.

In an ideal world, the folders would still be named by UUIDs. But instead of the folder names representing the recording names, these recording names would either be stored in some local database or as a metadata file that's associated with those UUIDs. While this would not solve the inefficiency mentioned in the previous paragraph, this solution would allow multiple recordings to be titled the same (e.g. three `Recording 2` can exist, or something as generic can be implemented like `Untitled Recording` for all new records). This way, we can eliminate the need to implement something in the previous paragraph altogether and avoid the issues that comes with it.

So while we can go with the second approach, I currently do not have enough time given my growing internship work. As such, I used UUID for each new folder/recording name as a trade-off. Also, Gemini doesn't pay rip. If this is a dealbreaker, I can try to see what can be done.

Reply:
Implemented the second solution I suggested to resolve race conditions and allow duplicate recording names.